### PR TITLE
Technical/add locations m apping

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**

### CODE REVIEW

**Review by oleg1std — APPROVED** [review]

File: HappyTravel.StaticDataMapper.Api/Services/Workers/LocationMapper.cs [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+            var localityZones = new List<KeyValuePair<(string CountryCode,string LocalityName), LocalityZone>>();
+            int skip = 0;
+            do
+            {
+                localityZones = await (from z in _context.LocalityZones
+                        join l in _context.Localities on z.LocalityId equals l.Id
+                        join c in _context.Countries on l.CountryId equals c.Id
+                        select new
+                            KeyValuePair<(string CountryCode, string LocalityName), LocalityZone>(
+                                // TODO: check
```
Comment (kirillta) [inline HappyTravel.StaticDataMapper.Api/Services/Workers/LocationMapper.cs:337]: Checked?

Comment (hasmikharutyunyan) [reply HappyTravel.StaticDataMapper.Api/Services/Workers/LocationMapper.cs:337]: Not yet, all testings are suspended, will be done after multiLanguage models integration here.

File: HappyTravel.StaticDataMapper.Api/Infrastructure/CountriesCache.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+using System.Threading.Tasks;
+using FloxDc.CacheFlow;
+using FloxDc.CacheFlow.Extensions;
+using HappyTravel.StaticDataMapper.Data.Models;
+
+namespace HappyTravel.StaticDataMapper.Api.Infrastructure
+{
+    public class CountriesCache : ICountriesCache
+    {
+        public CountriesCache(IDoubleFlow doubleFlow)
```
[the remark this answers is not in the document]
Comment (hasmikharutyunyan) [reply HappyTravel.StaticDataMapper.Api/Infrastructure/CountriesCache.cs:11]: Maybe not, I thoght about that what to use memory, redis or db, and maybe after testing will be changed to memory cache, for countries it possible I think

File: HappyTravel.StaticDataMapper.Api/Services/Workers/AccommodationMapper.cs [hunk 3]
```diff
@@ -40,14 +42,21 @@ namespace HappyTravel.StaticDataMapper.Api.Services.Workers
     public class AccommodationMapper : IAccommodationMapper
     {
         public AccommodationMapper(NakijinContext context, IAccommodationsTreesCache treesCache,
-            ILoggerFactory loggerFactory, 
+            ILoggerFactory loggerFactory,
             // TODO: change for generic Options model
-            IOptions<AccommodationsPreloaderOptions> options)
+            IOptions<AccommodationsPreloaderOptions> options, ILocationNameNormalizer locationNameNormalizer,
+            ICountriesCache countriesCache, ILocalitiesCache localitiesCache, ILocalityZonesCache localityZonesCache,
```
[the remark this answers is not in the document]
Comment (hasmikharutyunyan) [reply HappyTravel.StaticDataMapper.Api/Services/Workers/AccommodationMapper.cs:48]: You mean to have one `LocationsCache` class for all caches?

**Pull request merged**